lib/core/include/scipp/core/multi_index.h

```diff
@@ -343,15 +221,15 @@ template <scipp::index N> class MultiIndex {
   std::array<scipp::index, NDIM_OP_MAX + 1> m_shape = {};
   /// Total number of dimensions.
   scipp::index m_ndim{0};
-  /// Number of dense dimensions, i.e. same as m_ndim when not binned,
-  /// else number of dims in bins.
-  scipp::index m_inner_ndim{0};
-  /// Index of dim referred to by bin indices to distinguish, e.g., 2D bins
-  /// slicing along first or second dim.
-  /// -1 if not binned.
-  scipp::index m_nested_dim_index{-1};
-  /// Parameters of the currently loaded bins.
-  std::array<BinIterator, N> m_bin = {};
+  /// Scale factor for bin size (1 unless bins are multi-dim).
+  scipp::index m_bin_size_scale{-1};
+  /// Start/stop indices for binned args, used for translating m_data_index to
+  /// index for the bin content.
+  std::array<const std::pair<scipp::index, scipp::index> *, N> m_indices = {};
```

Comment:
```suggestion
  std::array<std::span<const std::pair<scipp::index, scipp::index>>, N> m_indices = {};
```
I was confused about the indexing in `get_deref_binned` because I initially thought that you used a pointer as a workaround for storing references here.

Reply:
What is the `span` doing? I don't understand your suggestion.

Reply:
This is conceptually an array of arrays. So storing is as an `std::array` of pointers is ambiguous.

Reply:
Hmm, true. We would pay some (most like negligible) memory for the end of the spans.